sasmodels/models/octahedron_truncated.c

```diff
@@ -0,0 +1,226 @@
+#include <math.h>
+#include <stdio.h> 
+static double
+form_volume(double length_a, double b2a_ratio, double c2a_ratio, double t)
+{
+//octehedron volume formula
+// length_a is the half height along the a axis of the octahedron
+    return (4./3.) * length_a * (length_a*b2a_ratio) * (length_a*c2a_ratio)*(1.-3*(1.-t)*(1.-t)*(1.-t));
+}
+
+static double
+Iq(double q,
+    double sld,
+    double solvent_sld,
+    double length_a,
+    double b2a_ratio,
+    double c2a_ratio,
+    double t)
+{
+    const double length_b = length_a * b2a_ratio;
+    const double length_c = length_a * c2a_ratio;
+
+
+   //Integration limits to use in Gaussian quadrature
+    const double v1a = 0.0;
+    const double v1b = M_PI_2;  //theta integration limits
+    const double v2a = 0.0;
+    const double v2b = M_PI_2;  //phi integration limits
+
+    double outer_sum = 0.0;
+    for(int i=0; i<GAUSS_N; i++) {
+        const double theta = 0.5 * ( GAUSS_Z[i]*(v1b-v1a) + v1a + v1b );
+        double sin_theta, cos_theta;
+        SINCOS(theta, sin_theta, cos_theta);
+
+        double inner_sum = 0.0;
+        for(int j=0; j<GAUSS_N; j++) {
+            double phi = 0.5 * ( GAUSS_Z[j]*(v2b-v2a) + v2a + v2b );
+            double sin_phi, cos_phi;
+            SINCOS(phi, sin_phi, cos_phi);
+
+            //HERE: Octahedron formula
+            const double Qx = q * sin_theta * cos_phi;
+    	    const double Qy = q * sin_theta * sin_phi;
+    	    const double Qz = q * cos_theta;
+    	    const double qx = Qx * length_a;
+    	    const double qy = Qy * length_b;
+    	    const double qz = Qz * length_c;
+
+            const double AA = 1./((qy*qy-qz*qz)*(qy*qy-qx*qx))*((qy-qx)*sin(qy*(1.-t)-qx*t)+(qy+qx)*sin(qy*(1.-t)+qx*t))+
```

Comment:
(qy*qy - qz*qz) probably needs to be taken care of. Otherwise we can end up in NaN/Inf scenario